app/services/i18n.js

```diff
@@ -0,0 +1,19 @@
+import { merge } from '@ember/polyfills';
+import I18nService from 'ember-i18n/services/i18n';
+
+export default I18nService.extend({
+    _globals: {},
+
+    t(key, data = {}) {
+        const mutableData = merge(this.get('_globals'), data);
+        return this._super(key, mutableData);
+    },
+
+    addGlobal(key, value) {
+        this.set(`_globals.${key}`, value);
+    },
+
+    addGlobals(globals) {
+        this.set('_globals', merge(this.get('_globals'), globals));
```

Comment:
Instead of all this "global" stuff, it might make sense to use [`addTranslations`](https://github.com/jamesarosen/ember-i18n/wiki/Doc:-Defining-Translations#defining-translations-at-runtime), which gets us approximately the same thing in a way we don't have to maintain.